### PR TITLE
Last call result

### PR DESCRIPTION
**Pull request by kubaplas**

### CODE REVIEW

File: odra-core/Cargo.toml [hunk 1]
```diff
@@ -5,4 +5,5 @@ edition = "2021"
 
 [dependencies]
 odra-types = { path = "../types" }
-casper-event-standard = "0.4.0"
\ No newline at end of file
+casper-event-standard = "0.4.0"
+bytes = "1.5.0"
```
Comment (kpob) [inline odra-core/Cargo.toml]: Do we use it?

**Review by kpob — APPROVED** [review]

File: odra-core/Cargo.toml (hunk 1 above)
Comment (kubaplas) [reply odra-core/Cargo.toml]: Removed

File: examples2/src/erc20.rs [hunk 2]
```diff
@@ -629,25 +777,159 @@ mod tests {
...
+        // Test call result
+        assert_eq!(env.get_events_count(&erc20.address), 2);
+
+        let call_result = env.last_call_result();
+        assert!(call_result.result.is_ok());
+        assert_eq!(call_result.contract_address, erc20.address);
+        assert_eq!(call_result.caller, alice);
+        assert_eq!(
+            call_result.result.unwrap(),
+            Bytes::from(().to_bytes().unwrap())
```
Comment (kpob) [inline examples2/src/erc20.rs]: Would be easier to convert an empty vec into Bytes.
```rust
vec![].into()
```

File: examples2/src/erc20.rs [hunk 3]
```diff
@@ -629,25 +777,159 @@ mod tests {
...
+        assert_eq!(call_result.caller, alice);
+        assert_eq!(call_result.events.get(&erc20.address).unwrap(), &vec![]);
+
+        // cross call
+        pobcoin.transfer(erc20.address, 100.into());
+        erc20.cross_transfer(pobcoin.address, alice, 50.into());
+        let call_result = env.last_call_result();
+
+        assert_eq!(
+            call_result.events.get(&pobcoin.address).unwrap(),
```
Comment (kpob) [inline examples2/src/erc20.rs]: Not really like the way you access events. I would create a getter that clones the event. Now you must pass a reference as an expected value.
```rust
call_result.get_events(&pobcoin.address)
```

**Review by kpob — APPROVED** [review]

File: examples2/src/erc20.rs [hunk 4]
```diff
@@ -629,25 +777,159 @@ mod tests {
...
+        assert_eq!(call_result.contract_address, erc20.address);
+        assert_eq!(call_result.caller, alice);
+        assert_eq!(
+            call_result.result.unwrap(),
+            vec![].into()
+        );
+        assert_eq!(
+            call_result.events.get(&erc20.address).unwrap(),
+            &vec![Bytes::from(
+                Transfer {
```
Comment (kpob) [inline examples2/src/erc20.rs:808]: Would be nice to have a direct conversion from events to Bytes and write just
```rust
Transfer {}.into()
```
Out of  this PR scope ofc,

File: examples2/src/erc20.rs (hunk 2 above)
Comment (kubaplas) [reply examples2/src/erc20.rs]: Done

File: examples2/src/erc20.rs (hunk 4 above)
Comment (kubaplas) [reply examples2/src/erc20.rs:808]: https://github.com/odradev/odra/issues/251

File: examples2/src/erc20.rs (hunk 3 above)
Comment (kubaplas) [reply examples2/src/erc20.rs]: Done as requested

**Pull request merged**